### PR TITLE
use in query instead of several or

### PR DESCRIPTION
**Pull request by yanns**

### CODE REVIEW

**Comment by yanns** [comment]
@hajoeichler @emmenko please review

File: src/coffee/stockimport.coffee [hunk 1]
```diff
@@ -220,13 +220,18 @@ class StockImport
...
-      _.each uniqueStocksToProcessBySku, (s) =>
+      predicate = "sku in ("
+      skus = _.map uniqueStocksToProcessBySku, (s) =>
         @_summary.emptySKU++ if _.isEmpty s.sku
         # TODO: query also for channel?
-        ie.where("sku = \"#{s.sku}\"")
+        "\"#{s.sku}\""
+      predicate += _.reduce skus, (s1, s2) =>
+        s1 + ", " + s2
+      predicate += ")"
```
Comment (emmenko) [inline src/coffee/stockimport.coffee]: @yanns Rewrite it like this

```coffeescript
skus = _.map uniqueStocksToProcessBySku, (s) =>
  @_summary.emptySKU++ if _.isEmpty s.sku
  # TODO: query also for channel?
  "\"#{s.sku}\""
predicate = "sku in (#{skus.join(', ')})"
```

Comment (yanns) [inline src/coffee/stockimport.coffee]: :+1: 
I do it

Comment (yanns) [inline src/coffee/stockimport.coffee]: https://github.com/sphereio/sphere-stock-import/commit/b1d0dca8ffbe70813356bca3b9d39c3a959a0602

**Comment by hajoeichler** [comment]
So, you remove the sortBySku in this case and we are back to sortById.
Does sort have any impact here on performance?

**Comment by yanns** [comment]
I've tested both (sort by SKU and sort by ID) and have not seen any difference.

File: src/coffee/stockimport.coffee [hunk 2]
```diff
@@ -220,13 +220,15 @@ class StockImport
...
-      _.each uniqueStocksToProcessBySku, (s) =>
+      skus = _.map uniqueStocksToProcessBySku, (s) =>
         @_summary.emptySKU++ if _.isEmpty s.sku
         # TODO: query also for channel?
-        ie.where("sku = \"#{s.sku}\"")
+        "\"#{s.sku}\""
+      predicate = "sku in (#{skus.join(', ')})"
 
-      ie.sort('sku').fetch()
+      @client.inventoryEntries.all()
```
Comment (hajoeichler) [inline src/coffee/stockimport.coffee]: Do we need `all()` here?

Comment (emmenko) [inline src/coffee/stockimport.coffee]: Yes, you want to get **all** inventories that match the predicate (not just one page)

**Pull request merged**